uni/lib/model/providers/session_provider.dart

```diff
@@ -52,11 +54,21 @@ class SessionProvider extends StateProviderNotifier {
         passwordController.clear();
         await acceptTermsAndConditions();
 
+        errorMessage = null;
         updateStatus(RequestStatus.successful);
       } else {
+        errorMessage = 'Credenciais inválidas';
+
+        //Check if password expired
+        final String responseHtml = await NetworkRouter.loginInSigarra(username, password, faculties);
+        if(isPasswordExpired(responseHtml)){
+          errorMessage = "A palavra-passe expirou";
+        }
         updateStatus(RequestStatus.failed);
       }
     } catch (e) {
+      // No internet connection or server down
+      errorMessage = "Verifica a tua ligação à internet";
       updateStatus(RequestStatus.failed);
```

Comment:
Maybe here throw an exception and handle it in the completer, it makes the code cleaner, you can throw different exceptions depending on the context of the error and not dependant on a string (that might make translation difficult in the future).